website/docs/cookbooks/introduction/formatting.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Formatting
+sidebar_position: 5
+---
+
+import {ChainedSnippets} from "../../../src/components/MarkdownComponents.js";
+
+Scala CLI also lets you format your code using [Scalafmt](https://scalameta.org/scalafmt/) formatter.
+
+
+## Basic usage
+
+This section describes how to use Scala CLI to format your code using default rules. For advance usage and all availible flags, please see next section.
+
+### Check
+
+To check the format of all your sources in the current directory:
+```bash
+scala-cli fmt --check .
+```
+
+This command will check if the code is formatted correctly according to the default rules.
+If it is not, it will return a diff of the changes that need to be made to match formatting rules.
+
+### Format
+
+If you want to format all your code, you can use the following command:
+```bash
+scala-cli fmt .
+```
+This command will format all your sources in the current directory according to the default rules.
+
+### Example
+
+Let's take a simple `HelloWorld` application:
+```scala title=HelloWorld.scala
+// use scala 3.3.0
+@main def hello()={println("Hello, World!")}
+def someLongMethodMoreThan80Characters():Unit={println("This is a long method")
+println("Another print here?")}
+```
+
+You can notice that this implementations is written with long lines, no indentation, no spaces and no new lines.
+It is hard to distinguish where one statement ends and another begins, which makes it hard to read.
+
+<ChainedSnippets>
+
+Let's check the format of this code:
+```bash
+scala-cli fmt --check .
+```
```

Comment:
the `docs-tests` CI job seems to be failing on this snippet... you're likely expecting the `--check` to fail here, so perhaps you need to add the `fail` keyword:
````suggestion
```bash fail
scala-cli fmt --check .
```
````
to check locally, try running:
```bash
./mill -i 'docs-tests[]'.test 'sclicheck.DocTests.cookbook introduction/formatting'
```